server/src/main/java/io/druid/guice/http/JettyHttpClientModule.java

```diff
@@ -116,6 +117,9 @@ public HttpClient get()
 
       httpClient.setIdleTimeout(config.getReadTimeout().getMillis());
       httpClient.setMaxConnectionsPerDestination(config.getNumConnections());
+      final QueuedThreadPool queuedThreadPool = new QueuedThreadPool(config.getNumMaxThreads());
```

Comment:
Unless there's some guarantee that the pool lifecycle properly manages the thread in the overall lifecycle